core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java

```diff
@@ -73,7 +73,8 @@ public class ThriftTransportPool {
 
   private ThriftTransportPool(LongSupplier maxAgeMillis) {
     this.maxAgeMillis = maxAgeMillis;
-    this.checkThread = Threads.createThread("Thrift Connection Pool Checker", () -> {
+    // TODO KEVIN RATHBUN all this does is perform some resource cleanup, so may not be critical.
+    this.checkThread = Threads.createNonCriticalThread("Thrift Connection Pool Checker", () -> {
```

Comment:
all this does is perform some resource cleanup, so may not be critical.

Reply:
Critical or not isn't really about "importance". It's about whether the system is resilient enough to re-kick the operation later. Even if this just cleans up resources, it would be "critical" in this sense if the system never tried to monitor or restart this thread if it were to die.

Reply:
Looks like the ThriftTransportPool is only created in the ClientContext.  Maybe we could make ServerContext and ClientContext do something different that leads to different behavior for client vs server for this thread.  Seems like halting the client would be bad.  

Maybe [this](https://github.com/apache/accumulo/blob/bdbe4f5173d49ddc347fc58be48796fe15a863b1/core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java#L329) should somehow be used on the client side for this case, but nto sure.

Reply:
I'll do this as a follow on. Seems like the changes for this won't be as straightforward as other places. Would probably be easier to review as separate PR too.